pkg/util/restoptions/configgetter.go

```diff
@@ -108,7 +107,7 @@ func (g *configRESTOptionsGetter) GetRESTOptions(resource unversioned.GroupResou
 		// TODO: choose destination group/version based on input group/resource
 		// TODO: Tune the cache size
 		groupVersion := unversioned.GroupVersion{Group: "", Version: g.masterOptions.EtcdStorageConfig.OpenShiftStorageVersion}
-		g.etcdHelper = etcdstorage.NewEtcdStorage(etcdClient, kapi.Codecs.LegacyCodec(groupVersion), g.masterOptions.EtcdStorageConfig.OpenShiftStoragePrefix, false, genericserveroptions.DefaultDeserializationCacheSize)
+		g.etcdHelper = etcdstorage.NewEtcdStorage(etcdClient, kapi.Codecs.LegacyCodec(groupVersion), g.masterOptions.EtcdStorageConfig.OpenShiftStoragePrefix, false, 1000)
```

Comment:
Access to a resource is almost certainly correlated to the update of a
resource - very, very few of our resources do anything except GET - UPDATE
- GET.

On Tue, Aug 30, 2016 at 5:59 PM, Seth Jennings <notifications@github.com>
wrote:

> In pkg/util/restoptions/configgetter.go
> <https://github.com/openshift/origin/pull/10719#discussion_r76887875>:
>
> > @@ -108,7 +107,7 @@ func (g *configRESTOptionsGetter) GetRESTOptions(resource unversioned.GroupResou
> >  		// TODO: choose destination group/version based on input group/resource
> >  		// TODO: Tune the cache size
> >  		groupVersion := unversioned.GroupVersion{Group: "", Version: g.masterOptions.EtcdStorageConfig.OpenShiftStorageVersion}
> > -		g.etcdHelper = etcdstorage.NewEtcdStorage(etcdClient, kapi.Codecs.LegacyCodec(groupVersion), g.masterOptions.EtcdStorageConfig.OpenShiftStoragePrefix, false, genericserveroptions.DefaultDeserializationCacheSize)
> > +		g.etcdHelper = etcdstorage.NewEtcdStorage(etcdClient, kapi.Codecs.LegacyCodec(groupVersion), g.masterOptions.EtcdStorageConfig.OpenShiftStoragePrefix, false, 1000)
>
> With the way the cache works, it will tend to fill up with resource types
> whose resourceVersions advance more rapidly. So if we sized the cache that
> way, there is no guarantee that the distribution of resource types in the
> cache would be even.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10719/files/5787df6f68b4ae442b248e9d0623fbda04c71562#r76887875>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p9gmBmd0fgNu-YZsqpkFTogRDoPAks5qlKfTgaJpZM4Jw21S>
> .
>